reframe/core/schedulers/slurm.py

```diff
@@ -451,11 +454,28 @@ def poll(self, *jobs):
             t_start = time.strftime(
                 '%F', time.localtime(min(job.submit_time for job in jobs))
             )
-            completed = _run_strict(
-                f'sacct -S {t_start} -P '
-                f'-j {",".join(job.jobid for job in jobs)} '
-                f'-o jobid,state,exitcode,end,nodelist'
-            )
+            try:
+                completed = _run_strict(
+                    f'sacct -S {t_start} -P '
+                    f'-j {",".join(job.jobid for job in jobs)} '
+                    f'-o jobid,state,exitcode,end,nodelist'
+                )
+                # Reset the retry counter if the command succeeds
+                self._sacct_retries = 0
+            except SpawnedProcessError as e:
+                if self._sacct_retries < self._max_sacct_retries:
+                    self._sacct_retries += 1
```

Comment:
I would invert here the check and the increment using the failure count logic:

```suggestion
                self._num_sacct_failures += 1
                if self._num_sacct_failures > self._max_sacct_failures:
```